service/backend/pchain/block.go

```diff
@@ -0,0 +1,431 @@
+				Index: blockIndex,
+				Hash:  block.BlockID.String(),
+			},
+			ParentBlockIdentifier: &types.BlockIdentifier{
+				Index: blockIndex - 1,
+				Hash:  block.ParentID.String(),
+			},
+			Timestamp:    block.Timestamp,
+			Transactions: transactions,
+		},
+	}
+
+	return resp, nil
+}
+
+// BlockTransaction implements the /block/transaction endpoint.
+func (b *Backend) BlockTransaction(ctx context.Context, request *types.BlockTransactionRequest) (*types.BlockTransactionResponse, *types.Error) {
+	isGenesisBlockRequest, err := b.isGenesisBlockRequest(ctx, request.BlockIdentifier.Index, request.BlockIdentifier.Hash)
+	if err != nil {
+		return nil, service.WrapError(service.ErrClientError, err)
+	}
+
+	var transactions []*types.Transaction
+
+	if isGenesisBlockRequest {
+		_, transactions, err = b.getGenesisBlockAndTransactions(ctx, request.NetworkIdentifier)
+		if err != nil {
+			return nil, service.WrapError(service.ErrInternalError, err)
+		}
+	} else {
+		block, err := b.getBlockDetails(ctx, request.BlockIdentifier.Index, request.BlockIdentifier.Hash)
+		if err != nil {
+			return nil, service.WrapError(service.ErrClientError, err)
+		}
+
+		transactions, err = b.parseTransactions(ctx, request.NetworkIdentifier, block.Txs)
+		if err != nil {
+			return nil, service.WrapError(service.ErrInternalError, err)
+		}
+	}
+
+	for i := range transactions {
+		transaction := transactions[i]
+		if transaction.TransactionIdentifier.Hash == request.TransactionIdentifier.Hash {
+			return &types.BlockTransactionResponse{
+				Transaction: transaction,
+			}, nil
+		}
+	}
+
+	return nil, service.ErrTransactionNotFound
+}
+
+func (b *Backend) parseTransactions(
+	ctx context.Context,
+	networkIdentifier *types.NetworkIdentifier,
+	txs []*txs.Tx,
+) ([]*types.Transaction, error) {
+	dependencyTxs, err := b.fetchDependencyTxs(ctx, txs)
+	if err != nil {
+		return nil, err
+	}
+
+	parser, err := b.newTxParser(ctx, networkIdentifier, dependencyTxs)
+	if err != nil {
+		return nil, err
+	}
+
+	transactions := []*types.Transaction{}
+	for _, tx := range txs {
+		err = common.InitializeTx(b.codecVersion, b.codec, tx)
+		if err != nil {
+			return nil, errTxInitialize
+		}
+
+		t, err := parser.Parse(tx.ID(), tx.Unsigned)
+		if err != nil {
+			return nil, err
+		}
+
+		transactions = append(transactions, t)
+	}
+	return transactions, nil
+}
+
+func (b *Backend) fetchDependencyTxs(ctx context.Context, txs []*txs.Tx) (map[string]*pmapper.DependencyTx, error) {
+	dependencyTxIDs := []ids.ID{}
+
+	for _, tx := range txs {
+		inputTxsIds, err := pmapper.GetDependencyTxIDs(tx.Unsigned)
+		if err != nil {
+			return nil, err
+		}
+		dependencyTxIDs = append(dependencyTxIDs, inputTxsIds...)
+	}
+
+	dependencyTxChan := make(chan *pmapper.DependencyTx, len(dependencyTxIDs))
+	eg, ctx := errgroup.WithContext(ctx)
+
+	dependencyTxs := make(map[string]*pmapper.DependencyTx)
+	for i := range dependencyTxIDs {
+		txID := dependencyTxIDs[i]
```

Comment:
An alternative is
```
for _, txID := range dependencyTxIDs {
	txID := txID
```
the second line needed to avoid capturing from the goroutine